sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala

```diff
@@ -1929,6 +1929,18 @@ class DataSourceV2SQLSuite
     }
   }
 
+  test("SPARK-30289: DSv2 `InMemoryTable` partitioning should not accept nested columns") {
+    val t1 = s"${catalogAndNamespace}tbl"
+    withTable(t1) {
+      val e = intercept[IllegalArgumentException] {
+        sql(s"CREATE TABLE $t1 (nested struct<id:bigint, data:string>) " +
```

Comment:
we don't support `CREATE TABLE USING fileSourceV2` now,  we only need to fix `InMemoryTable`.